accepted/0000-caasp-nodes.md

```diff
@@ -0,0 +1,277 @@
+
+1. Temporarily disable minion blackout from Uyuni/SUSE Manager using a Salt formula.
+2. The user then executes any action that was intended for the CaaS Platform node.
+3. Finally, minion blackout is enabled again:
+    - Manually by the user
+    - Automatically, as a hook after every action issued by the user
+    - Automatically, in a timely fashion (e.g. after 10 minutes)
+
+The functionality would be used also for troubleshooting and installing troubleshooting tools and issuing commands.
+
+The risk of breaking the cluster by issuing any forbidden operation is still present while the blackout is disabled.
+
+Value of this solution:
+* Easy and quick to implement
+* The user can still break its cluster by issuing any forbidden action while the cluster is not in the blackout
+* The Workflow for the user is different and mechanical, can lead to errors
+
+#### Uyuni/SUSE Manager action locking
+
+The idea is that Uyuni/SUSE Manager must restrict the following features:
+
+- Reboot a system:
+  - If the targeted minion is a `caasp_node` system type, the action for rebooting a system:
+    - Must be hidden from the UI and the corresponding action must fail: code paths `/systems/details/RebootSystem`, `/systems/ssm/misc/RebootSystem`, `ActionChainManager#scheduleRebootAction`, `ActionChainManager#scheduleRebootActions`
+    - The corresponding action must fail if scheduled via XMLRPC: `SystemHandler#scheduleReboot`
+- Package upgrade:
+  - If the targeted minion is a `caasp_node` system type, the action for upgrade a package:
+    - Must be hidden from the UI and the corresponding action must fail: code paths `/systems/details/packages/UpgradableList`, `systems/ssm/PackageUpgrade`, `ActionChainManager#schedulePackageUpgrade`, `ActionChainManager#schedulePackageUpgrades`
+    - The corresponding action must fail if scheduled via XMLRPC: [`SystemHandler#schedulePackageInstall`]
+- Package removal:
+  - If the targeted minion is a `caasp_node` system type, the action for removing a package:
+    - Must be hidden from the UI and the corresponding action must fail: `/systems/details/packages/RemoveConfirm`, `systems/ssm/PackageRemove`, `ActionChainManager#schedulePackageRemoval`, `ActionChainManager#schedulePackageRemoval`
+    - The corresponding action must fail if scheduled via XMLRPC: `SystemHandler#schedulePackageRemove`
+- Patch apply:
+  - If the targeted minion is a `caasp_node` system type, the action for applying a patch:
+    - Must be hidden from the UI and the corresponding action must fail: `/systems/details/ErrataList.do`, `/systems/ssm/ListErrata`, `ErrataManager#applyErrata`
+    - The corresponding action must fail if scheduled via XMLRPC: [`SystemHandler#scheduleApplyErrata`]
+
+Value of this solution:
+* This workflow does not constitute a radical change for a user
+* Change does not involve Salt, the user can still break its cluster by issuing Salt commands
+* Requires changes in Uyuni/SUSE Manager core code
+
```

Comment:
I am not sure if both **Uyuni/SUSE Manager action locking** & **Minion blackout** methods are mutually exclusive in its current form. We will probably need both in one way or the other.

IIUC, if we decide for **Minion blackout**, we will still need to hide the pages from UI & adjust XMLRPC API points to throw an error in such cases. It wouldn't be efficient to actually reach at salt level and then return the error, we should not even allow to create action/serveraction instance in DB in such cases.

 Similarly, if we only go for "**Uyuni/SUSE Manager action locking**" proposal, then we left susemanager exposed as you mentioned above and customer can do whatever she wants using salt commands directly.

Reply:
Yes, very much agreed. I will push a correction here.